### PR TITLE
`[PRD]` | `[Issue #2530]` | Hierarchy voting strategies display

### LINKED ISSUE #2530: Add voting schema “ERC-20”, “ERC-721”, “Multisig” to the listed subdaos
https://www.notion.so/hulking-cosmos-3e8/Q4-2024-QA-Requirements-13e2ee9c5bc280a88458e7a935d17488?pvs=4#13f2ee9c5bc2801bbacad5d7e7fd659e

Probably means the hierarchy page's subdao nodes should indicate what type they are. Correct if wrong

### PR DESCRIPTION
**Pull request by Da-Colon**
Closes #2530 

So i'd like this PR -> #2574  -> #2566 to all be merged together and then merged into develop, This will keep the caching for organization page all the same for version one. 

## Screenshot

localhost_8888_hierarchy_dao=base_0x0c14aA2d34C0E13Bbe441963e4de2EeAef4abeD2(Desktop)

### CODE REVIEW

**Comment by Da-Colon** [comment]
@decentdao/design technically there can be multiple strategy 

Example: with current setup
image

**Comment by Da-Colon** [comment]
@nicolaus-sherrill We also spoke about Snapshot's Component. Its there. but needs some work. I have a seperate branch I'll open to handle that work

File: src/components/DaoHierarchy/DaoHierarchyNode.tsx
Comment (Da-Colon) [inline src/components/DaoHierarchy/DaoHierarchyNode.tsx:1]: `getVotingStrategies` and `getGovernanceTypes` was ripped from `useVotingStrategyAddress` and `useGovernanceContracts` and reduced down for the need here.

File: src/components/DaoHierarchy/DaoHierarchyNode.tsx [hunk 1]
```diff
@@ -51,6 +58,77 @@ export function DaoHierarchyNode({
...
+      if (!votingStrategies) {
+        throw new Error('No voting strategies found');
+      }
+
+      if (!publicClient) {
+        throw new Error('Public Client is not set!');
+      }
+
+      let governanceTypes: DaoHierarchyStrategyType[] = [];
+
```
Comment (Da-Colon) [inline src/components/DaoHierarchy/DaoHierarchyNode.tsx:107]: I did remove a section of code look for LockRelease and ERC20Wrapper contracts, didn't seem needed here.

File: src/components/DaoHierarchy/DaoHierarchyNode.tsx [hunk 2]
```diff
@@ -51,6 +58,77 @@ export function DaoHierarchyNode({
...
+          } = votingStrategy;
+          if (isLinearVotingErc20) {
+            governanceTypes.push('ERC-20');
+          } else if (isLinearVotingErc721) {
+            governanceTypes.push('ERC-721');
+          } else if (isLinearVotingErc20WithHatsProposalCreation) {
+            governanceTypes.push('ERC-20');
+          } else if (isLinearVotingErc721WithHatsProposalCreation) {
+            governanceTypes.push('ERC-721');
+          }
```
Comment (Da-Colon) [inline src/components/DaoHierarchy/DaoHierarchyNode.tsx:124]: It was discussed in standup that we would just keep in ERC-20~ without any additional info (like whitelisting).

**Comment by nicolaus-sherrill** [comment]
@Da-Colon good shout on multiple strategies. Wasn't aware of that. I'll update the designs so we have a variation captured.

re: hover & neutral 3 — I'm aware of that. It looks like the pattern was already in place w/ the Snapshot space chip, so I left it alone. Not a huge deal to me right now — if there's flack we can swap that chip color to our White Alpha 04 color so that it always stays visible.

**Comment by DarksightKellar** [comment]
ERC20 showing twice

Screenshot 2024-11-28 at 12 11 44

**Review by mudrila — APPROVED** [review]

**Comment by Da-Colon** [comment]
> @Da-Colon ERC20 showing twice. Bug?
> 
> Screenshot 2024-11-28 at 12 11 44

Weird let me see if I accidently pushed my example code.

**Comment by Da-Colon** [comment]
> @Da-Colon ERC20 showing twice. Bug?
> 
> Screenshot 2024-11-28 at 12 11 44

What DAO? Things are appearing okay for me?

**Comment by DarksightKellar** [comment]
@Da-Colon Sorry here's the DAO address: 0x536ccCeA88ea2720F455D846E2c05673306b7642

**Review by DarksightKellar — APPROVED** [review]

**Comment by parkermccurley** [comment]
@Da-Colon please make "MULTISIG" => "Multisig"

**Pull request merged**